examples/audiopanning/main.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	_ "image/png"
+	"io"
+	"log"
+	"math"
+	"time"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/audio"
+	"github.com/hajimehoshi/ebiten/audio/vorbis"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	raudio "github.com/hajimehoshi/ebiten/examples/resources/audio"
+	"github.com/hajimehoshi/ebiten/examples/resources/images"
+)
+
+const (
+	screenWidth  = 320
+	screenHeight = 240
+	sampleRate   = 22050
+
+	introLengthInSecond = 5
+	loopLengthInSecond  = 4
+)
+
+var img *ebiten.Image
+
+var audioContext *audio.Context
+
+func init() {
+	var err error
+	audioContext, err = audio.NewContext(sampleRate)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type Game struct {
+	player    *audio.Player
+	panstream *StereoPanStream
+
+	// panning goes from -1 to 1
+	// -1: 100% left channel, 0% right channel
+	// 0: 100% both channels
+	// 1: 0% left channel, 100% right channel
+	panning float64
+
+	r    float64
```

Comment:
`r` should an integer representing a count in order to avoid numeric errors. Maybe renaming it to `count` sounds fine. So the logic will be like this:

```go
g.count++
r := float64(g.count) * ((1.0 / 60.0) * 2 * math.Pi) * 0.1
g.xpos = (float64(screenWidth) / 2) + math.Cos(r)*(float64(screenWidth)/2)
```